### PR TITLE
[FrameworkBundle][Serializer] Move SerializerPass to the Serializer

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Part of #21284

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -44,20 +44,21 @@
...
         "symfony/security-core": "~3.2",
         "symfony/security-csrf": "~2.8|~3.0",
-        "symfony/serializer": "~2.8|~3.0",
+        "symfony/serializer": "~3.3",
         "symfony/translation": "~2.8|~3.0",
         "symfony/templating": "~2.8|~3.0",
         "symfony/validator": "~3.2",
         "symfony/yaml": "~3.2",
-        "symfony/property-info": "~2.8|~3.0",
+        "symfony/property-info": "~3.1",
```
Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Why this change?

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Because we change the framework's serializer requirement to 3.3, and the serializer in 3.3 requires this. The build with low deps was failing otherwise:

> 1) Symfony\Bundle\FrameworkBundle\Tests\Functional\SerializerTest::testDeserializeArrayOfObject
Error: Class 'phpDocumentor\Reflection\FileReflector' not found

See https://travis-ci.org/symfony/symfony/jobs/191965990

Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Well, then it means that the serializer component should have the proper requirement or the proper conflict in place

Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: It should be a conflict because the Serializer doesn't require PropertyInfo.

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: See https://github.com/symfony/symfony/pull/21309

Comment (dunglas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: I'll open a PR for that.

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: @dunglas done in #21309

Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: reverted thanks to #21309

**Comment by chalasr** [comment]
@nicolas-grekas Is there any reason for adding this to the 3.x milestone but https://github.com/symfony/symfony/pull/21375 to the 3.3 one? They are about fixing the same ticket so if there is something wrong to you here, please tell me :)

**Comment by chalasr** [comment]
Updated this to make the serializer id and encoder/normalizer tags configurable through the constructor, defaulting to the ones used in the framework.

**Comment by chalasr** [comment]
But, isn't a BC break to add a parent class with a new constructor to the deprecated pass?

**Comment by chalasr** [comment]
Do we want to move forward on this for 3.3? It would be nice to have a thought, to know if it's worth working on other relevant passes listed in the ticket using the same approach.

**Comment by xabbuh** [comment]
IMO it makes sense to move them.

**Comment by dunglas** [comment]
👍

**Comment by fabpot** [comment]
Thank you @chalasr.
